core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3077,14 +3080,13 @@ private void loadConfig() throws IOException {
             // load from disk
             cfg.unmarshal(Jenkins.this);
         }
-
         try {
             checkRawBuildsDir(buildsDir);
             setBuildsAndWorkspacesDir();
         } catch (InvalidBuildsDir invalidBuildsDir) {
             throw new IOException(invalidBuildsDir);
         }
-
+        configLoaded = true;
```

Comment:
was unsure if this should go before setting the builds/workspaces directory or after.  (we should be up enough that a save can be called as soon as `cfg.unmashal(Jenkins.this)` has returned, but at the same time the builds/workspaces directories may be used by something too.